### PR TITLE
1972 - drafting @tag documentation.

### PR DESCRIPTION
**Pull request by jeffjakub**
Creating a draft PR to iterate on the `@tag` directive spec.
This PR relates to: https://github.com/apollographql/federation/issues/1972

### CODE REVIEW

File: docs/source/federation-spec.md [hunk 1]
```diff
@@ -413,3 +413,41 @@ type User @key(fields: "id") {
...
 }
 ```
+
+### `@tag`
+
+```graphql
+directive @tag(from: String!) on FIELD_DEFINITION | INTERFACE | OBJECT | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION
+```
+
+The `@tag` directive is used to provide a mechanism for applying arbitrary string metadata to the fields and types of a schema. This metadata is potentially useful throughout the schema’s lifecycle, including, but not limited to, processing, static analysis, and documentation.
```
Comment (dariuszkuc) [inline docs/source/federation-spec.md:423]: should we also link to https://www.apollographql.com/docs/studio/contracts as that is the primary use case?

File: docs/source/federation-spec.md [hunk 2]
```diff
@@ -413,3 +413,41 @@ type User @key(fields: "id") {
...
+
+```graphql
+schema
+  @core(feature: "https://specs.apollo.dev/core/v0.2")
```
Comment (dariuszkuc) [inline docs/source/federation-spec.md:427]: should this be `@link`ed instead? i.e. nowhere else in the docs we mention `@core` directive in our examples

Comment (clenfest) [reply docs/source/federation-spec.md:427]: Yes, should be @link. @core was the old way. Also, we should get the @tag from the federation CoreFeature.

```@link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@tag"])```

File: docs/source/federation-spec.md [hunk 3]
```diff
@@ -413,3 +413,41 @@ type User @key(fields: "id") {
...
+
+### `@tag`
+
+```graphql
+directive @tag(from: String!) on FIELD_DEFINITION | INTERFACE | OBJECT | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION
+```
+
+The `@tag` directive is used to provide a mechanism for applying arbitrary string metadata to the fields and types of a schema. This metadata is potentially useful throughout the schema’s lifecycle, including, but not limited to, processing, static analysis, and documentation.
+
+```graphql
```
Comment (clenfest) [inline docs/source/federation-spec.md:425]: I would add a sentence, saying that @tag will be propagated up to the supergraph automatically (as Phil mentioned in the @inaccessible spec)

**Pull request closed without merging**